columnflow/calibration/cms/egamma.py

```diff
@@ -0,0 +1,581 @@
+# coding: utf-8
+
+"""
+Egamma energy correction methods.
+Source: https://twiki.cern.ch/twiki/bin/view/CMS/EgammSFandSSRun3#Scale_And_Smearings_Correctionli
+"""
+
+from __future__ import annotations
+
+import abc
+import functools
+import itertools
+import law
+from dataclasses import dataclass, field
+
+from columnflow.calibration import Calibrator, calibrator
+from columnflow.calibration.util import ak_random
+from columnflow.util import maybe_import, InsertableDict
+from columnflow.columnar_util import (
+    set_ak_column, flat_np_view, ak_copy, optional_column,
+)
+from columnflow.types import Any
+
+ak = maybe_import("awkward")
+np = maybe_import("numpy")
+
+
+# helper
+set_ak_column_f32 = functools.partial(set_ak_column, value_type=np.float32)
+
+
+@dataclass
+class EGammaCorrectionConfig:
+    correction_set: str = "Scale"
+    corrector_kwargs: dict[str, Any] = field(default_factory=dict)
+
+
+class egamma_scale_corrector(Calibrator):
+
+    with_uncertainties = True
+    """Switch to control whether uncertainties are calculated."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    @abc.abstractmethod
```

Comment:
There's also `abc.abstractproperty` for this. Also applies to a couple places below (e.g. in egamma_resolution_corrector).